New glob loaders find resources outside of eventcatalog

### I tried this:

After updating from 2.29.4 to 2.30.7 we noticed a strange error when building the catalog.

### This happened:

When running `npm run build` we received the following error:
```
10:03:32 [ERROR] [vite] ✗ Build failed in 6.01s
[@mdx-js/rollup] Expected a closing tag for `<url>` (23:30-23:35) before the end of `paragraph`
file: .../eventcatalog/custom-generator/node_modules/source-map-url/changelog.mdx:23:9
```

While its not clear to me what exactly the error is, the changelog.md(x) file mentioned does exist but is something outside of our eventcatalog.

### I expected this:

It should not look for files outside the scope of eventcatalog.

### Is there a workaround?

_No response_

### Anything else?

_No response_

### EventCatalog Version

2.30.7

### Node.js Version

22

### Platform(s)

_No response_

### Community Notes

<!-- Please keep this note for the community -->
* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

--- a/.changeset/light-planes-work.md
+++ b/.changeset/light-planes-work.md
@@ -1,0 +1,5 @@
+---
+"@eventcatalog/core": patch
+---
+
+fix(core): astro loaders no longer read node_modules

--- a/eventcatalog/src/content.config.ts
+++ b/eventcatalog/src/content.config.ts
@@ -14,7 +14,7 @@ export const projectDirBase = (() => {
 
 const pages = defineCollection({
   loader: glob({
-    pattern: ['**/pages/*.(md|mdx)'],
+    pattern: ['**/pages/*.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
   }),
   schema: z
@@ -43,7 +43,7 @@ const resourcePointer = z.object({
 
 const changelogs = defineCollection({
   loader: glob({
-    pattern: ['**/changelog.(md|mdx)'],
+    pattern: ['**/changelog.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
   }),
   schema: z.object({
@@ -133,7 +133,7 @@ const flowStep = z
 
 const flows = defineCollection({
   loader: glob({
-    pattern: ['**/flows/*/index.(md|mdx)', '**/flows/*/versioned/*/index.(md|mdx)'],
+    pattern: ['**/flows/*/index.(md|mdx)', '**/flows/*/versioned/*/index.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
     generateId: ({ data }) => {
       return `${data.id}-${data.version}`;
@@ -203,7 +203,7 @@ const flows = defineCollection({
 
 const events = defineCollection({
   loader: glob({
-    pattern: ['**/events/*/index.(md|mdx)', '**/events/*/versioned/*/index.(md|mdx)'],
+    pattern: ['**/events/*/index.(md|mdx)', '**/events/*/versioned/*/index.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
     generateId: ({ data, ...rest }) => {
       return `${data.id}-${data.version}`;
@@ -222,7 +222,7 @@ const events = defineCollection({
 
 const commands = defineCollection({
   loader: glob({
-    pattern: ['**/commands/*/index.(md|mdx)', '**/commands/*/versioned/*/index.(md|mdx)'],
+    pattern: ['**/commands/*/index.(md|mdx)', '**/commands/*/versioned/*/index.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
     generateId: ({ data }) => {
       return `${data.id}-${data.version}`;
@@ -241,7 +241,7 @@ const commands = defineCollection({
 
 const queries = defineCollection({
   loader: glob({
-    pattern: ['**/queries/*/index.(md|mdx)', '**/queries/*/versioned/*/index.(md|mdx)'],
+    pattern: ['**/queries/*/index.(md|mdx)', '**/queries/*/versioned/*/index.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
     generateId: ({ data }) => {
       return `${data.id}-${data.version}`;
@@ -309,7 +309,7 @@ const domains = defineCollection({
 
 const channels = defineCollection({
   loader: glob({
-    pattern: ['**/channels/*/index.(md|mdx)', '**/channels/*/versioned/*/index.(md|mdx)'],
+    pattern: ['**/channels/*/index.(md|mdx)', '**/channels/*/versioned/*/index.(md|mdx)', '!**/node_modules/**'],
     base: projectDirBase,
     generateId: ({ data }) => {
       return `${data.id}-${data.version}`;

--- a/scripts/build-ci.js
+++ b/scripts/build-ci.js
@@ -27,4 +27,9 @@ execSync(`cross-env NODE_ENV=CI PROJECT_DIR=${projectDIR} CATALOG_DIR=${catalogD
 // Type check
 execSync(`pnpm exec astro check --minimumSeverity error --root ${catalogDir}`, {
   stdio: 'inherit',
+  env: {
+    ...process.env,
+    CATALOG_DIR: catalogDir,
+    PROJECT_DIR: projectDIR,
+  },
 });

--- a/scripts/build-ci.js
+++ b/scripts/build-ci.js
@@ -28,7 +28,6 @@ execSync(`cross-env NODE_ENV=CI PROJECT_DIR=${projectDIR} CATALOG_DIR=${catalogD
 execSync(`pnpm exec astro check --minimumSeverity error --root ${catalogDir}`, {
   stdio: 'inherit',
   env: {
-    ...process.env,
     CATALOG_DIR: catalogDir,
     PROJECT_DIR: projectDIR,
   },

